### PR TITLE
Add Tynn::StatusHandler plugin.

### PR DESCRIPTION
**Pull request by frodsan**
@soveran This solves what we talked the other day. Let me know what you think :-)

### CODE REVIEW

File: lib/tynn/not_found.rb [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-        res.finish
-      end
+    def self.setup(app)
+      app.plugin(Tynn::StatusHandler)
+      app.handle(404, :not_found)
+    end
 
+    # @private
+    module InstanceMethods
       def not_found
```
Comment (moonglum) [inline lib/tynn/not_found.rb]: Does it do nothing by default?

Comment (frodsan) [inline lib/tynn/not_found.rb]: I guess I can check if the method is defined and remove this. But I don't think it should be doing something, the status is 404 by the time is executed.

Comment (frodsan) [inline lib/tynn/not_found.rb]: Also, I'm thinking if this plugin should be included by default?

Comment (moonglum) [inline lib/tynn/not_found.rb]: Hm, you're right :smile: That's the better solution!

Comment (moonglum) [inline lib/tynn/not_found.rb]: The plugin is needed when you want to treat your 404 pages differently than just returning the 404 status code, right? I don't think we need this by default :smile:

**Pull request closed without merging**